db/database.go

```diff
@@ -880,13 +880,15 @@ func VacuumAttachments(bucket base.Bucket) (int, error) {
 }
 
 // backgroundTask runs task at the specified time interval in its own goroutine until the changeCache is stopped.
-func (context *DatabaseContext) backgroundTask(name string, task func(), interval time.Duration) {
+func (context *DatabaseContext) backgroundTask(name string, task func() error, interval time.Duration) {
```

Comment:
Discussed with @bbrks, and while it might be an exciting opportunity for me to make all kinds of mistakes with Contexts, maybe hold off on unifying further for now. Maybe when we have a 3rd type of background task?